### PR TITLE
feat(guardian-api): add pagination to groups/users endpoints

### PR DESCRIPTION
**Pull request by amacar**
### What changes are being made?
Add pagination and orderBy (groups only) to groups and users endpoints.

CU-86u7n1

### Why are these changes necessary?
To allow a paginated view in GUI.

## Checklist

- [ ] Documentation _(if necessary)_
- [x] Tests _(if necessary)_
- [x] Ready to be merged

### CODE REVIEW

**Comment by kristjank** [comment]
Task linked: [CU-86u7n1 feat(guardian-api): add pagination to users/groups endpoints](https://app.clickup.com/t/86u7n1)

File: packages/guardian-api/src/resources/users.ts [hunk 1]
```diff
@@ -1,30 +1,5 @@
-import { Contracts } from "@arkecosystem/core-api";
-import { Container } from "@arkecosystem/core-kernel";
+import { Interfaces } from "@protokol/guardian-transactions";
 
-@Container.injectable()
-export class UserResource implements Contracts.Resource {
```
Comment (KovacZan) [inline packages/guardian-api/src/resources/users.ts:5]: Why did you remove Contracts.Resource class, each response should have one, so we follow the same convention as in nfts and in core-api

File: packages/guardian-api/src/resources/groups.ts [hunk 2]
```diff
@@ -1,27 +1,11 @@
-import { Contracts } from "@arkecosystem/core-api";
-import { Container } from "@arkecosystem/core-kernel";
+import { Schemas } from "@arkecosystem/core-api";
+import Joi from "@hapi/joi";
 
-@Container.injectable()
-export class GroupResource implements Contracts.Resource {
```
Comment (KovacZan) [inline packages/guardian-api/src/resources/groups.ts:5]: The same goes here

File: packages/guardian-api/src/resources/users.ts (hunk 1 above)
Comment (amacar) [reply packages/guardian-api/src/resources/users.ts:5]: Based on this PR: https://github.com/ArkEcosystem/core/pull/3931 resources aren't transformer classes anymore.

Comment (KovacZan) [reply packages/guardian-api/src/resources/users.ts:5]: ok thanks for letting me know

**Review by KovacZan — APPROVED** [review]
